packages/esm-patient-search-app/src/patient-search.resource.tsx

```diff
@@ -0,0 +1,99 @@
+import { useCallback, useMemo } from 'react';
+import { openmrsFetch, FetchResponse, useConfig } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+import useSWRInfinite from 'swr/infinite';
+import { SearchedPatient, FHIRPatientSearchResponse, FHIRPatientType, PatientSearchResponse } from './types';
+
+const v =
+  'custom:(patientId,uuid,identifiers,display,' +
+  'patientIdentifier:(uuid,identifier),' +
+  'person:(gender,age,birthdate,birthdateEstimated,personName,addresses,display,dead,deathDate),' +
+  'attributes:(value,attributeType:(name)))';
+
+export function usePatientSearchPaginated(
+  searchTerm: string,
+  searching: boolean = true,
+  resultsToFetch: number,
+  page: number,
+  customRepresentation: string = v,
+) {
+  const config = useConfig();
+  let url = `/ws/rest/v1/patient?q=${searchTerm}&v=${customRepresentation}&limit=${resultsToFetch}&totalCount=true`;
+  if (config.includeDead) {
+    url += `&includeDead=${config?.includeDead}`;
+  }
+  if (page > 1) {
+    url += `&startIndex=${(page - 1) * resultsToFetch}`;
+  }
+
+  const { data, isValidating, error } = useSWR<
+    FetchResponse<{ results: Array<SearchedPatient>; links: Array<{ rel: 'prev' | 'next' }>; totalCount: number }>
+  >(searching ? url : null, openmrsFetch);
+
+  const results: {
+    data: Array<SearchedPatient>;
+    isLoading: boolean;
+    fetchError: any;
+    hasMore: boolean;
+    loadingNewData: boolean;
+    totalResults: number;
+  } = useMemo(
+    () => ({
+      data: data?.data?.results,
+      isLoading: !data?.data && !error,
+      fetchError: error,
+      hasMore: data?.data?.links?.some((link) => link.rel === 'next'),
+      loadingNewData: isValidating,
+      totalResults: data?.data?.totalCount,
+    }),
+    [data, isValidating, error],
+  );
+
+  return results;
```

Comment:
For these hooks wouldn't you also be interested in returning `isLoading` and `error` also?

Reply:
I have it in the results object, checkout the line 43.
Thanks!